pkg/k8shandler/common.go

```diff
@@ -758,8 +759,8 @@ func EnforceNetworkPolicy(namespace string, client client.Client, ownerRef []met
 
 	err := client.Create(context.TODO(), &policy)
 	if err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("Could not create network policy: %v", err)
+		if !apierrors.IsAlreadyExists(errors.Root(err)) {
```

Comment:
The k8s errors package is not very good. It's not using the go 1.13 [errors.Is](https://golang.org/pkg/errors/#Is). Instead, it is using its own type assertion so it will only evaluate to true if `err` is unwrapped. `errors.Root` unwraps an error until it finds the root cause of the error so that `IsAlreadExists` can match accordingly. The k8s API needs to be updated to use `errors.Is` instead of their own hand-rolled type assertions.